packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -56,33 +55,29 @@ export class WorkspaceMainImpl implements WorkspaceMain {
         this.inPluginFileSystemWatcherManager = new InPluginFileSystemWatcherManager(this.proxy, container);
 
         workspaceService.roots.then(roots => {
-            this.roots = roots;
-            this.notifyWorkspaceFoldersChanged();
+            this.notifyWorkspaceFoldersChanged(roots);
+        });
+
+        workspaceService.onWorkspaceChanged(roots => {
+            this.notifyWorkspaceFoldersChanged(roots);
         });
     }
 
-    notifyWorkspaceFoldersChanged(): void {
-        if (this.roots && this.roots.length) {
-            const folders = this.roots.map((root: FileStat, index: number) => {
-                const uri = Uri.parse(root.uri);
-                const path = new Path(uri.path);
-                return {
-                    uri: uri,
-                    name: path.base,
-                    index: index
-                };
-            });
+    notifyWorkspaceFoldersChanged(roots: FileStat[]): void {
+        if (this.isAnyRootChanged(roots) === false) {
```

Comment:
Maybe use `!this.isAnyRootChanged(roots)`

Reply:
I don't mind to use here `!` as well, but I prefer this form because it is easier to read for me.